android/src/main/java/com/swmansion/reanimated/layoutReanimation/SharedTransitionManager.java

```diff
@@ -0,0 +1,400 @@
+import android.util.Log;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.ViewParent;
+import com.facebook.react.bridge.ReactContext;
+import com.facebook.react.uimanager.IllegalViewOperationException;
+import com.facebook.react.uimanager.PixelUtil;
+import com.facebook.react.uimanager.ViewGroupManager;
+import com.facebook.react.views.view.ReactViewGroup;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import javax.annotation.Nullable;
+
+public class SharedTransitionManager {
+  private final AnimationsManager mAnimationsManager;
+  private NativeMethodsHolder mNativeMethodsHolder;
+  private final List<View> mAddedSharedViews = new ArrayList<>();
+  private final Map<Integer, View> mSharedTransitionParent = new HashMap<>();
+  private final Map<Integer, Integer> mSharedTransitionInParentIndex = new HashMap<>();
+  private boolean mIsSharedTransitionActive;
+  private final Map<Integer, Snapshot> mSnapshotRegistry = new HashMap<>();
+  private final Map<Integer, View> mCurrentSharedTransitionViews = new HashMap<>();
+  private View mTransitionContainer;
+  private final List<View> mRemovedSharedViews = new ArrayList<>();
+
+  public SharedTransitionManager(AnimationsManager animationsManager) {
+    mAnimationsManager = animationsManager;
+  }
+
+  protected void notifyAboutNewView(View view) {
+    mAddedSharedViews.add(view);
+  }
+
+  protected void notifyAboutRemovedView(View view) {
+    mRemovedSharedViews.add(view);
+  }
+
+  @Nullable
+  protected View getTransitioningView(int tag) {
+    return mCurrentSharedTransitionViews.get(tag);
+  }
+
+  protected void viewsDidLayout() {
+    setupSharedTransitionForViews(mAddedSharedViews, true);
+    mAddedSharedViews.clear();
+  }
+
+  protected void onViewsRemoving(int[] tagsToDelete) {
+    if (tagsToDelete != null) {
+      visitTreeForTags(tagsToDelete, false);
+
+      setupSharedTransitionForViews(mRemovedSharedViews, false);
+      for (View removedSharedView : mRemovedSharedViews) {
+        visitTree(removedSharedView.getId(), true);
+      }
+      mRemovedSharedViews.clear();
+
+      visitTreeForTags(tagsToDelete, true);
+    }
+  }
+
+  protected void doSnapshotForTopScreenViews(ViewGroup stack) {
+    int screensCount = stack.getChildCount();
+    if (screensCount > 0) {
+      View firstStackChild = stack.getChildAt(0);
+      if (firstStackChild instanceof ViewGroup) {
+        View screen = ((ViewGroup) firstStackChild).getChildAt(0);
+        visitTreeAndMakeSnapshot(screen);
+      } else {
+        Log.e("[Reanimated]", "Unable to recognize screen on stack.");
+      }
+    }
+  }
+
+  protected void setNativeMethods(NativeMethodsHolder nativeMethods) {
+    mNativeMethodsHolder = nativeMethods;
+  }
+
+  private void setupSharedTransitionForViews(List<View> sharedViews, boolean withNewElements) {
```

Comment:
```suggestion
  private void startSharedTransitionForViews(List<View> sharedViews, boolean withNewElements) {
```
since this actually _starts_ the transition, right?